tests/core/data_model/test_timestamp.py

```diff
@@ -0,0 +1,46 @@
+# Copyright The Caikit Authors
```

Comment:
Assuming we only use timestamp internally, set by trainers, we probably don't need to test `to_json`, `from_json` and `to_dict`

Reply:
good point though, I could create a few `@dataobjects` in these tests that contain timestamps to ensure the json/dict stuff actually works